.github/workflows/upgrade-tests.yaml

```diff
@@ -0,0 +1,89 @@
+name: Upgrade Tests
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  upgrade-tests:
+    runs-on: ubuntu-latest
+    timeout-minutes: 90
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+
+    env:
+      BRANCH_NAME: ${{ github.event.inputs.branch || 'master' }}
+
+    steps:
+      - name: Checkout noobaa-core (master/input branch)
+        uses: actions/checkout@v4
+        with:
+          repository: 'noobaa/noobaa-core'
+          path: 'noobaa-core'
+          ref: ${{ env.BRANCH_NAME }}
+
+      - name: Deploy minikube
+        run: |
+          cd ./noobaa-core
+          sudo bash ./.travis/deploy_minikube.sh
+
+      - name: Build noobaa
+        run: |
+          cd ./noobaa-core
+          make noobaa NOOBAA_TAG=noobaa-core:${{ env.BRANCH_NAME || 'master' }}
+
+      - name: Checkout noobaa-core (latest)
+        uses: actions/checkout@v4
+        with:
+          repository: 'noobaa/noobaa-core'
+          path: 'noobaa-core'
+
+      - name: Build noobaa (latest)
+        run: |
+          cd ./noobaa-core
+          make noobaa NOOBAA_TAG=noobaa-core:upgrade-tests
+
+      - name: Checkout noobaa-operator
+        uses: actions/checkout@v4
+        with:
+          repository: 'noobaa/noobaa-operator'
+          path: 'noobaa-operator'
+
+      - name: Change settings for k8s and minikube
+        run: |
+          sudo mv /root/.kube /root/.minikube $HOME
+          sudo chown -R $USER $HOME/.kube $HOME/.minikube
+          sed "s/root/home\/$USER/g" $HOME/.kube/config > tmp; mv tmp $HOME/.kube/config
+
+      - name: Build operator
+        run: |
+          cd ./noobaa-operator
+          make all
+
+      - name: Install noobaa system (from master)
+        run: |
+          cd ./noobaa-operator
+          ./build/_output/bin/noobaa-operator crd create
+          ./build/_output/bin/noobaa-operator operator install
+          ./build/_output/bin/noobaa-operator system create  \
+          --db-resources='{ "limits": {"cpu": "200m","memory": "2G"}, "requests": {"cpu": "200m","memory": "2G"}}' \
+          --core-resources='{ "limits": {"cpu": "200m","memory": "1G"}, "requests": {"cpu": "200m","memory": "1G"}}' \
+          --endpoint-resources='{ "limits": {"cpu": "200m","memory": "1G"}, "requests": {"cpu": "200m","memory": "1G"}}' \
+          --noobaa-image='noobaa-core:${{ env.BRANCH_NAME || 'master' }}'
```

Comment:
In the step you wrote: "Install noobaa system (from master)" - did you mean master in the core repo, right? Then why `--noobaa-image='noobaa-core:${{ env.BRANCH_NAME || 'master' }}'`? I think it should be `noobaa-core:upgrade-tests` (which according to the steps represents "Build noobaa (latest)".
Are you sure the system should be installed in the latest version?

Reply:
I have also added pull request referece in latest build (in step 4) in my last commit, it would work I think.

Reply:
Removed the pull request reference as we don't need it I can see the latest code logs inside the upgrade tests pipeline, this proves that we checked out to latest code. Thanks

Reply:
I think I still don't understand it - I think that the title should be updated "Install noobaa system (from master)" It is not from master as we passed --noobaa-image='noobaa-core:${{ env.BRANCH_NAME || 'master' }}'